subsys/net/ip/net_context.c

```diff
@@ -1977,6 +1982,7 @@ static int context_sendto(struct net_context *context,
 		}
 	}
 
+skip_alloc:
 	if (IS_ENABLED(CONFIG_NET_OFFLOAD) &&
 	    net_if_is_ip_offloaded(net_context_get_iface(context))) {
 		ret = context_write_data(pkt, buf, len, msghdr);
```

Comment:
I probably missed something, but could this be reached with pkt == NULL? and can context_write_data accept a NULL pkt?

Reply:
It may also be good to say in the commit message what throughput increase this change gives

Reply:
Oh boy, offloading again... Yes, you're right - we should not skip the packet allocation if the interface if offloaded, I've added an extra check for that.